src/runtime/current_thread/task_executor.rs

```diff
@@ -24,34 +26,73 @@ impl TaskExecutor {
     ///
     /// For more details, see the [module level](index.html) documentation.
     pub fn current() -> TaskExecutor {
-        TaskExecutor {
-            inner: TaskExecutor02::current(),
-        }
+        TaskExecutor { _p: () }
     }
 
     /// Spawn a `futures` 0.1 future onto the current `CurrentThread` instance.
     pub fn spawn_local(
         &mut self,
-        future: impl Future01<Item = (), Error = ()> + 'static,
+        future: Box<dyn Future01<Item = (), Error = ()>>,
     ) -> Result<(), executor_01::SpawnError> {
-        let future = Box::pin(future.compat().map(|_| ()));
-        self.spawn_local_std(future).map_err(map_spawn_err)
+        self.spawn_local_std(future.compat().map(|_| ()))
     }
 
-    /// Spawn a `std::future` future onto the current `CurrentThread` instance.
+    /// Spawn a `std::futures`  future onto the current `CurrentThread`
+    /// instance.
     pub fn spawn_local_std(
         &mut self,
-        future: Pin<Box<dyn Future<Output = ()>>>,
-    ) -> Result<(), executor_02::SpawnError> {
-        self.inner.spawn_local(future)
+        future: impl Future<Output = ()> + 'static,
+    ) -> Result<(), executor_01::SpawnError> {
+        if let Some(idle) = Runtime::reserve_idle() {
+            tokio_02::task::spawn_local(idle.with(future));
+            Ok(())
+        } else {
+            Err(executor_01::SpawnError::shutdown())
+        }
+    }
+
+    /// Spawn a `futures` 0.1 future onto the current `CurrentThread` instance,
+    /// returning a `JoinHandle` that can be used to await its result.
+    ///
+    /// **Note** that futures spawned in this manner do not "count" towards
+    /// keeping the runtime active for [`shutdown_on_idle`], since they are paired
+    /// with a `JoinHandle` for  awaiting their completion. See [here] for
+    /// details on shutting down the compatibility runtime.
+    ///
+    /// [`shutdown_on_idle`]: struct.Runtime.html#method.shutdown_on_idle
+    /// [here]: ../index.html#shutting-down
+    ///
+    /// # Panics
+    ///
+    /// This function panics if there is no current runtime, or if the current
+    /// runtime is not a current-thread runtime.
+    pub fn spawn_handle<T: 'static, E: 'static>(
+        &mut self,
+        future: Box<dyn Future01<Item = T, Error = E>>,
```

Comment:
same question here.

Reply:
this one should probably just be impl future, since it isn't a drop-in replacement for an 0.1 fn